aws/resource_aws_xray_sampling_rule.go

```diff
@@ -0,0 +1,212 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/xray"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsXraySamplingRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsXraySamplingRuleCreate,
+		Read:   resourceAwsXraySamplingRuleRead,
+		Update: resourceAwsXraySamplingRuleUpdate,
+		Delete: resourceAwsXraySamplingRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"rule_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"resource_arn": {
```

Comment:
This does not allow `"*"`, which is the default value used by AWS

Reply:
If `"*"` is the only exception, then we should be able to do the following with [`validation.Any()`](https://godoc.org/github.com/hashicorp/terraform/helper/validation#Any):

```go
ValidateFunc: validation.Any(
	validateArn,
	validation.StringInSlice([]string{"*"}),
),
```

Otherwise, perfectly fine to ignore validation for now!

Reply:
I will ignore validation for now as the resource should accept ARNs such as  `arn:aws:lambda:*:*:function:*-dev`